containers/src/Kokkos_DualView.hpp

```diff
@@ -1186,12 +1203,11 @@ void deep_copy(
   }
 }
 
-template <class ExecutionSpace, class DT, class DL, class DD, class DM,
-          class ST, class SL, class SD, class SM>
+template <class ExecutionSpace, class DT, class... DProps, class ST, class... SProps>
 void deep_copy(
     const ExecutionSpace& exec,
-    DualView<DT, DL, DD, DM> dst,  // trust me, this must not be a reference
```

Comment:
Comment about why this is ok to delete that comment and not trust whoever wrote that.

Reply:
Well, my first feeling was that this code was wrong because it was calling `modify_host`/`modify_device` on a (shallow) copy of `dst`. But as the `modified_flags` member is a (shared) view, it seems this is still OK.

Still I was not "trusting" such a comment like *trust me, this must not be a reference* (reminds me of "Trust me I'm an engineer") because to me this should work with a reference otherwise there must be other cases where taking a dual view by reference is dangerous.

So I wanted to put my hypothesis that in fact it works with a reference to the test.

On my side, it worked for OpenMP and Cuda (without UVM).

Reply:
I am leaning towards saying your change is fine.
My point was this needs to be discussed and not swept under the rug.  It looks like the comment has been there since the initial commit in this project when the code was extracted from Sandia's Trilinos repository.
cc @crtrott

Reply:
It was introduced in https://github.com/trilinos/trilinos/commit/0806efcf77af0df08b5c82b6e57342e08144dbf4 by @mhoemmen.